onnx/defs/function.cc

```diff
@@ -13,6 +13,41 @@ std::string InteralTensorNameGenerator(
   return new_name;
 }
 
+bool nodes_equal(const NodeProto& lhs, const NodeProto& rhs) {
+
+  if (!lhs.has_op_type() || !rhs.has_op_type()) {
+    return false;
+  }
+  if (lhs.op_type() != rhs.op_type()){
+    return false;
+  }
+  if(lhs.attribute_size()!=rhs.attribute_size()){
+    return false;
+  }
+  for(int i=0; i<lhs.attribute_size(); i++){
+    if(lhs.attribute(i).name()!=rhs.attribute(i).name()){
```

Comment:
attribute value should also be checked?

Reply:
and also domain? and doc string?

do you think it's better to check the two nodeproto's serialized string instead?

Reply:
Yes, I agree that `SerializeToString` method from Protobuf will work better than my custom function.